packages/sol/src/verification.ts

```diff
@@ -0,0 +1,389 @@
+  couponCode: string
+): FieldError.CouponCodeInvalid | true {
+  if (!COUPON_REGEX.test(couponCode)) return FieldError.CouponCodeInvalid;
+  return true;
+}
+export function isValidCouponDiscount(
+  couponDiscount: number
+):
+  | FieldError.CouponDiscountZeroOrNegative
+  | FieldError.CouponDiscountAbove100Percent
+  | true {
+  if (couponDiscount <= DISCOUNT_MIN_VALUE)
+    return FieldError.CouponDiscountZeroOrNegative;
+  if (couponDiscount > DISCOUNT_MAX_VALUE)
+    return FieldError.CouponDiscountAbove100Percent;
+  return true;
+}
+export function isValidCouponExpireDate(
+  expireDate: string
+):
+  | FieldError.CouponExpireDateBeforeToday
+  | FieldError.CouponExpireDateInvalid
+  | true {
+  const currentTime = new Date().getTime();
+  const expireTime = new Date(expireDate).getTime();
+
+  if (!Date.parse(expireDate)) return FieldError.CouponExpireDateInvalid;
+  if (currentTime > expireTime) return FieldError.CouponExpireDateBeforeToday;
+  return true;
+}
+
+export function isValidMessageText(
+  message: string
+):
+  | FieldError.MessageTextInvalidHTML
+  | FieldError.MessageTextTooLong
+  | FieldError.MessageTextTooShort
+  | true {
+  const text = getTextFromHTMLString(message);
+
+  if (!REGEXFORHTML.test(message)) return FieldError.MessageTextInvalidHTML;
+  if (text.length > 1000) return FieldError.MessageTextTooLong;
+  if (text.length < 1) return FieldError.MessageTextTooShort;
+  return true;
+}
+
+export function isValidWithdrawMinAmount(
+  minAmount: number,
+  maxAmount: number
+):
+  | FieldError.WithdrawMinAmountAboveMaxAmount
+  | FieldError.WithdrawMaxAmountZeroOrNegative
+  | FieldError.WithdrawMinAmountZeroOrNegative
+  | true {
+  if (minAmount <= 0) return FieldError.WithdrawMinAmountZeroOrNegative;
+  if (maxAmount <= 0) return FieldError.WithdrawMaxAmountZeroOrNegative;
+  if (minAmount > maxAmount) return FieldError.WithdrawMinAmountAboveMaxAmount;
+  return true;
+}
+export function isValidWithdrawMaxAmount(
+  maxAmount: number,
+  minAmount: number
+):
+  | FieldError.WithdrawMinAmountAboveMaxAmount
+  | FieldError.WithdrawMaxAmountZeroOrNegative
+  | FieldError.WithdrawMinAmountZeroOrNegative
+  | true {
+  if (maxAmount <= 0) return FieldError.WithdrawMaxAmountZeroOrNegative;
+  if (minAmount <= 0) return FieldError.WithdrawMinAmountZeroOrNegative;
+  if (minAmount > maxAmount) return FieldError.WithdrawMinAmountAboveMaxAmount;
+  return true;
+}
+export function isValidInvoiceReceiver(
+  receiver: string,
+  type: IInvoice.RecivingMethod,
+  bankName?: Bank
+) {
+  if (type === IInvoice.RecivingMethod.Wallet) {
+    if (!PHONE_NUMBER_REGEX.test(receiver))
+      return FieldError.PhoneNumberInValid;
+    return true;
+  }
+  if (type === IInvoice.RecivingMethod.Instapay) {
+    if (!INSTAPAY_REGEX.test(receiver)) return FieldError.InstapayIPAInvalid;
+    return true;
+  }
+  // if (type === IInvoice.RecivingMethod.Bank) {
+  //   if (bankName === Bank.Alex) {
+
+  //   }
+  // }
```

Comment:
I see that you want to test the banks that the user provided
what do you think if we did something like this:
```
if (type === IInvoice.RecivingMethod.Bank) {
    if (!bankName) return FieldError.BankNameEmpty;
    if (!Object.values(Bank).includes(bankName)) {
    return FieldError.BankNameInvalid;
    }
    return true
  }
```
Don't forget to implement `FieldError.BankNameEmpty` and `FieldError.BankNameInvalid`